### PR TITLE
[core] ModelPartIO accepts 'True', 'true' or 1 for bool variables in mdpa files

### PR DESCRIPTION
**Pull request by maceligueta**
This is just a proposal, nothing crucial, to accept other formats for bool variables.
It was triggered by an error in the GiD GUI, which was printing True instead of 1, but I thought this might be useful.
However, the implementation I came to was not as nice as intended, because I had to specialize a template function.
I am totally open to suggestions and rejections.

### CODE REVIEW

File: kratos/tests/auxiliar_files_for_python_unittest/mdpa_files/test_model_part_io_read.mdpa [hunk 1]
```diff
@@ -84,7 +84,7 @@ End NodalData
 
 Begin SubModelPart Inlets
 	Begin SubModelPartData
-	IS_RESTARTED 1
+	IS_RESTARTED True
```
Comment (philbucher) [inline kratos/tests/auxiliar_files_for_python_unittest/mdpa_files/test_model_part_io_read.mdpa:88]: can we have a check for each of them?
i.e. one for `1`, one for `True` and one for `true`?

(not sure if there are enough vars of type `bool` in the Core)

File: kratos/sources/model_part_io.cpp [hunk 2]
```diff
@@ -4495,6 +4495,19 @@ TValueType& ModelPartIO::ExtractValue(std::string rWord, TValueType & rValue)
     return rValue;
 }
 
+bool& ModelPartIO::ExtractValue(std::string rWord, bool & rValue)
+{
+    KRATOS_WATCH(rWord)
```
Comment (philbucher) [inline kratos/sources/model_part_io.cpp]:
```suggestion
```

File: kratos/sources/model_part_io.cpp [hunk 3]
```diff
@@ -4495,6 +4495,19 @@ TValueType& ModelPartIO::ExtractValue(std::string rWord, TValueType & rValue)
...
+
+    if (rWord == "1" || rWord == "true" || rWord == "True") {
+        rValue = true;
+    }
+
+    else rValue = false;
```
Comment (philbucher) [inline kratos/sources/model_part_io.cpp]:
```suggestion
    if (rWord == "1" || rWord == "true" || rWord == "True") {
        rValue = true;
    } else {
          rValue = false;
    }
```

File: kratos/tests/auxiliar_files_for_python_unittest/mdpa_files/test_model_part_io_read.mdpa (hunk 1 above)
Comment (maceligueta) [reply kratos/tests/auxiliar_files_for_python_unittest/mdpa_files/test_model_part_io_read.mdpa:88]: > (not sure if there are enough vars of type `bool` in the Core)
Let me check, that is exactly what I was afraid of

File: kratos/sources/model_part_io.cpp (hunk 2 above)
Comment (maceligueta) [reply kratos/sources/model_part_io.cpp]: Many thanks

File: kratos/tests/auxiliar_files_for_python_unittest/mdpa_files/test_model_part_io_read.mdpa (hunk 1 above)
Comment (philbucher) [reply kratos/tests/auxiliar_files_for_python_unittest/mdpa_files/test_model_part_io_read.mdpa:88]: https://github.com/KratosMultiphysics/Kratos/blob/b356c2b07c94bd23fe9f26a883df926e623b077b/kratos/includes/variables.h#L43-L45

**Review by philbucher — DISMISSED** [review]
ok from my side, I think this is a good solution

maybe wait a few days with merging to see if others have comments too

File: kratos/sources/model_part_io.cpp [hunk 4]
```diff
@@ -4495,6 +4495,18 @@ TValueType& ModelPartIO::ExtractValue(std::string rWord, TValueType & rValue)
...
 }
 
+bool& ModelPartIO::ExtractValue(std::string rWord, bool & rValue)
+{
+
+    if (rWord == "1" || rWord == "true" || rWord == "True") {
+        rValue = true;
+    } else {
+          rValue = false;
+    }
```
Comment (philbucher) [inline kratos/sources/model_part_io.cpp]:
```suggestion
    } else if (rWord == "0" || rWord == "false" || rWord == "False") {
        rValue = false;
    } else {
        KRATOS_ERROR << "Boolean argument could not be determined: " << rWord << std::endl;
    }
```

Probably even better what do you think?

Comment (maceligueta) [reply kratos/sources/model_part_io.cpp]: Certainly much better!

**Review by philbucher — APPROVED** [review]

**Pull request merged**